### ISSUE TITLE
Incompatible record as argument in MOS2

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1387]
**Reported by jmattsson on 20 Dec 2013 10:04 UTC**
The model `Modelica.Electrical.Spice3.Internal.MOS2` contains the following function calls:
```mo
    final parameter Spice3.Internal.Mos2.Mos2Calc
                                   c1=Spice3.Internal.Mos.mos2CalcInitEquations(
             p,
             C,
             vp,
             m) "Precalculated parameters";
    final parameter Spice3.Internal.Mos2.Mos2Calc
                                   c2=Spice3.Internal.Mos.mos2CalcCalcTempDependencies(
             p,
             C,
             vp,
             m,
             c1,
             m_type) "Precalculated parameters";
```
The parameter vp used in both calls is of type `Spice3.Internal.Mos2.Mos2ModelLineVariables`, but the corresponding arguments in the functions are both of type `Spice3.Internal.Mos.MosModelLineVariables`. The spec requires record expressions to have the same named elements to be type compatible (and that is required for function arguments). In this case, `Mos2ModelLineVariables` has 3 elements not present in `MosModelLineVariables`, and is thus not compatible with the inputs to the functions.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1387

**Comment by modelica-trac-importer** [comment]
**Comment by jmattsson on 17 Nov 2014 15:04 UTC**
Fixed in 6ac199ec67bc7bb5992d419fd9c00a5e987f7b68 as a part of #1245.

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 17 Nov 2014 18:46 UTC**
